onnx_xla/onnx_xla_interface.cc

```diff
@@ -267,3 +277,79 @@ ONNXIFI_SYMBOL_NAME(onnxReleaseGraph)(onnxGraph graph) {
   }
   ONNXIFI_CATCH_EXCPETION();
 }
+
+//Initialize event by creating EventControl object on the heap
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxInitEvent)(onnxBackend backend, onnxEvent* event) {
+  try {
+    if (!backend)  {
+      return ONNXIFI_STATUS_INVALID_BACKEND;
+    }
+    if (!event)  {
+      return ONNXIFI_STATUS_INVALID_POINTER;
+    }
+
+    *event = (onnxEvent) new EventControl();
+    return ONNXIFI_STATUS_SUCCESS;
+  }
+  ONNXIFI_CATCH_EXCPETION();
+
+
+}
+
+//Signal Event by changing the signalled boolean under mutex hold
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxSignalEvent)(onnxEvent event)  {
+  try {
+    auto *e = reinterpret_cast<EventControl *>(event);
```

Comment:
IMO, it is better to cast `event` after you validate that it is non-null. Current code is correct, though.

Reply:
Agreed - changed